contract_invoice_merge_by_partner/models/account_analytic_analysis.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# © 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.multi
+    def _recurring_create_invoice(self, automatic=False):
+        invoice_obj = self.env['account.invoice']
+        invoices = invoice_obj.browse(
+            super(PurchaseOrderLine, self)._recurring_create_invoice(automatic)
+        )
+        res = []
```

Comment:
As this is new API, you are receiving a recordset of invoices, and so, you must return another invoices recordset.